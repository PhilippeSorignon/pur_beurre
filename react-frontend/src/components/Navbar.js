import React from "react";
import { Link } from 'react-router-dom';

class Navbar extends React.Component {
  signout = () => {
    localStorage.removeItem('authToken');
  }

  render() {
    return (
      <nav
        className="navbar navbar-expand-lg navbar-light fixed-top py-3"
        id="mainNav"
      >
        <div className="container">
          <a className="navbar-brand js-scroll-trigger" href="/">
            <img
              src="/static/assets/img/logo_pur_beurre.png"
              className="mr-2"
              width="50px"
            />
            Pur Beurre
          </a>
          <button
            className="navbar-toggler navbar-toggler-right"
            type="button"
            data-toggle="collapse"
            data-target="#navbarResponsive"
            aria-controls="navbarResponsive"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon" />
          </button>
          {localStorage.authToken
            ? <div className="collapse navbar-collapse" id="navbarResponsive" data-testid="loggednavbar">
                <ul className="navbar-nav ml-auto my-2 my-lg-0">
                  <li className="nav-item">
                    <a className="nav-link js-scroll-trigger" href="/me">
                      <i className="fas fa-user" />
                    </a>
                  </li>
                  <li className="nav-item">
                    <a className="nav-link js-scroll-trigger" href="/products">
                      <i className="fas fa-carrot" />
                    </a>
                  </li>
                  <li className="nav-item">
                    <a className="nav-link js-scroll-trigger" onClick={this.signout}>
                      <i className="fas fa-sign-out-alt" />
                    </a>
                  </li>
                </ul>
              </div>
            : <div className="collapse navbar-collapse" id="navbarResponsive">
                <ul className="navbar-nav ml-auto my-2 my-lg-0">
                  <li className="nav-item">
                    <a className="nav-link js-scroll-trigger" href="/login">
                      Connexion
                    </a>
                  </li>
                  <li className="nav-item">
                    <a className="nav-link js-scroll-trigger" href="/signup">
                      Inscription
                    </a>
                  </li>
                </ul>
              </div>
          }

        </div>
      </nav>
    );
  }
}

export default Navbar;
