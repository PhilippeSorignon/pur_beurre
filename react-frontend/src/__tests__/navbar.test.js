import React from 'react';
import ReactDOM from 'react-dom';
import Navbar from '../components/Navbar';

import { render, cleanup, fireEvent, wait } from '@testing-library/react';
import renderer from "react-test-renderer";
import { LocalStorageMock } from '@react-mock/localstorage';

afterEach(cleanup);

const renderComponent = () =>
  render(
    <LocalStorageMock items={{ authToken: 'token' }}>
      <Navbar />
    </LocalStorageMock>
  );


it("Navbar renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Navbar />, div);
});

it("Navbar display login when user is anonymous", () => {
  const { getByText } = render(<Navbar />);

  expect(getByText('Connexion')).toBeTruthy();
});

it("Navbar matches snapshot", () => {
  const tree = renderer.create(<Navbar />).toJSON();
  expect(tree).toMatchSnapshot();
});
