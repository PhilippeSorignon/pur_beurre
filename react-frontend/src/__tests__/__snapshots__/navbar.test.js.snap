// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches snapshot 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light fixed-top py-3"
  id="mainNav"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand js-scroll-trigger"
      href="/"
    >
      <img
        className="mr-2"
        src="/static/assets/img/logo_pur_beurre.png"
        width="50px"
      />
      Pur Beurre
    </a>
    <button
      aria-controls="navbarResponsive"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler navbar-toggler-right"
      data-target="#navbarResponsive"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarResponsive"
    >
      <ul
        className="navbar-nav ml-auto my-2 my-lg-0"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link js-scroll-trigger"
            href="/login"
          >
            Connexion
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link js-scroll-trigger"
            href="/signup"
          >
            Inscription
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
